proxy/nginx.conf

```diff
@@ -91,12 +91,24 @@ http {
     set_real_ip_from 0.0.0.0/0;
     real_ip_header X-Forwarded-For;
     real_ip_recursive on;
-    set $allow false;
-    if ($remote_addr ~ "159.142.") {
-      set $allow true;
+    set $authy "";
+
+    # Determine if auth cookie is set
+    if ($cookie_auth_tkt) {
+      set $authy C;
+    }
+    if ($uri = "/user/saml2login") {
+      set $authy "${authy}L";
+    }
+    if ($uri = "/acs") {
+      set $authy "${authy}S";
+    }
+    if ($uri = "/api/action/status_show") {
+      set $authy "${authy}H";
     }
-    if ($allow = false) {
-      return 403;
+    # If not logged in and not trying to login, redirect
+    if ($authy = "") {
+      return 302 https://{{env "EXTERNAL_ROUTE"}};
```

Comment:
```suggestion
      return 302 https://{{env "PUBLIC_ROUTE"}};
```

Reply:
Let us use PUBLIC_ROUTE as much as possible for redirect. It is CDN if CDN is available, or same value as EXTERNAL_ROUTE if CDN not available.